### ISSUE TITLE
[explicit-function-return-type] Correct return type for React function components

### ISSUE THREAD

**Issue by flopes89** [typescript-eslint/typescript-eslint#814]
**Repro**

```JSON
{
  "extends": "plugin:@typescript-eslint/recommended"
}
```

```TS
const MyComp: React.FC<MyProps> = (props): ???? => {
   return (
      <div>Hello World</div>
   );
};
```

What's the correct/recommended return type for the function when this rule tells me that I am required to use one?

I would assume it's inferred, but that doesn't seem to happen?

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `1.13.0` |
| `@typescript-eslint/parser`        | `1.13.0` |
| `TypeScript`                       | `3.4.3` |
| `ESLint`                           | `5.16.0` |
| `node`                             | `10.15.3` |
| `npm`                              | `6.4.1` |

**Comment by bradzacher** [comment]
A few ways:
- if you use go to definition, you'll be able to find out the expected return type of the `React.FC`
- if you delete the `React.FC` type and mouse over the variable name, it'll show you the expected return type

Also please consider reading the rule docs, as you can use the `allowTypedFunctionExpressions` option to avoid having to explicitly type this function.

**Comment by flopes89** [comment]
I know about the options, I'm just wondering if it makes sense to have them in the recommended configuration like they are.

**Comment by bradzacher** [comment]
https://github.com/typescript-eslint/typescript-eslint/pull/729 and friends.

It's going to be turned on by default in v2.

**Comment by flopes89** [comment]
Aaah I see, perfect! Thanks :)
